posts/2020-11-20-JNDI-gRPC-200012.adoc

```diff
@@ -0,0 +1,389 @@
+Long form remote interfaces and homes `ejb/#<package.qualified.interface>`
+
+The component-id defaults to `<application-name>/<module-jar-name>/<ejb-name>`
+
+[#migration]
+=== Easier Migration for Enterprise Bean Applications
+
+Custom JNDI name support for enterprise beans provides an easier migration path for applications from other platforms (including WebSphere traditional).
+
+Prior to Java EE 6, the Enterprise Beans specification did not prescribe the JNDI names required for enterprise beans, so every platform provided platform specific default names and custom binding file formats. Since Liberty only supported the specification defined JNDI names, migrating applications from other platforms often requires code changes to modify the platform specific JNDI names to the newer specification defined names. Now, migration from other platforms is simplified because applications may be migrated without changing code, but instead migrating the other platform specific binding files to the new Liberty platform specific binding file format. In some cases, use of the new legacy default names may also allow applications to migrate to liberty without specifying custom JNDI names in a binding file.
+
+[#config]
+=== Application Configuration details
+
+Full details about the legacy default bindings provided, as well as the syntax for declaring custom JNDI names in the ibm-ejb-jar-bnd.xml file, including examples, may be found in link:https://www.ibm.com/support/knowledgecenter/SSEQTP_9.0.5/com.ibm.websphere.base.doc/ae/cejb_bindingsejbfp.html[this IBM Knowledge Center article].
+
+Custom bindings may be configured for an application in the following three locations.
+
+[customibm]
+==== Specify Custom bindings in ibm-ejb-jar-bnd.xml for EJB 3.x
+
+Following are examples of how to configure custom bindings for EJB 3.x beans in an EJB JAR module or WAR module in `ibm-ejb-jar-bnd.xml`
+
+Specify a binding per interface:
+
+
+[source, XML]
+----
+   <session name="NoInterceptorBasicStateless">
+      <interface class="com.ejbs.InventoryService" binding-name="ejb/Inventory"/>
+   </session>
+----
+
+Specify a component id (a prefix for default long form bindings)
+
+
+[source, XML]
+----
+   <session name="AccountServiceBean" component-id="Dept549/AccountProcessor"/>
+----
+
+Simple binding name (one name used for both local and remote)
+
+
+[source, XML]
+----
+   <session name="AccountServiceBean" simple-binding-name="ejb/AccountService"/>
+----
+
+Local and Remote home specific binding names
+
+
+[source, XML]
+----
+   <session name="AccountServiceBean" local-home-binding-name="ejblocal:AccountService"/>
+   <session name="AccountServiceBean" remote-home-binding-name="ejb/services/AccountService"/>
+----
+
+[customserver]
+==== Specify Custom bindings in server.xml
+
+Following is an example of how to configure custom bindings for EJB 3.x beans in an EJB JAR module or WAR module in `server.xml` in the `<application>` or `<ejbApplicationd>` elements:
+
```

Comment:
Change `<ejbApplicationd>` to `<ejbApplication>`

Reply:
Done